src/Data/TMap.hs

```diff
@@ -188,17 +188,17 @@ keys = F.keys
 {-# INLINE keys #-}
 
 -- | Return the list of keys by wrapping them with a user-provided function.
-keysWith :: (forall a. TypeRep a -> r) -> TMap -> [r]
+keysWith :: (forall a . TypeRep a -> r) -> TMap -> [r]
 keysWith = F.keysWith
 {-# INLINE keysWith #-}
 
 -- | Return the list of key-value pairs by wrapping them with a user-provided function.
-toListWith :: (forall a. Typeable a => a -> r) -> TMap -> [r]
-toListWith = F.toListWith
+toListWith :: (forall a . Typeable a => a -> r) -> TMap -> [r]
+toListWith f = F.toListWith (f . runIdentity)
```

Comment:
That was an unobvious and unexpected change 😮 I believe it has something to do with Impredicative Types but not sure. Good fix on this one!